### ISSUE TITLE
paramsSerializer is not working in v0.28.0

### ISSUE THREAD

**Issue by briwa** [axios/axios#6262]
### Describe the bug

Although there's a backport in #6227, in v0.28.0, the serializer isn't working as expected. This is because the v0.28.0 version of `buildUrl` ([code](https://github.com/axios/axios/blob/3b7635aefc842c05da0ec8c90e8bd09cb54616b8/lib/helpers/buildURL.js#L36-L40)) is simply only taking `options.encode`, not taking `options.serializer` into account:

https://github.com/axios/axios/blob/ce46346c0f5306b65023bba324c76fe5fe83eee8/lib/helpers/buildURL.js#L36-L44

This seems to be a refactor to match the v1.x version, because in v0.27.2 at least ([code](https://github.com/axios/axios/blob/v0.27.2/lib/helpers/buildURL.js#L28-L30)), it takes the `paramSerializer` function into account:

https://github.com/axios/axios/blob/bc733fec78326609e751187c9d453cee9bf1993a/lib/helpers/buildURL.js#L28-L30

In order for #6227 to actually work, there needs to be a backport similar to #5113 towards v0.x

### To Reproduce

- Set the version to v0.28.0
- Set the `paramsSerializer` function
- It won't work

### Code snippet

```js
// Simply following https://axios-http.com/docs/req_config

axios.create({
  paramsSerializer: function (params) {
    return Qs.stringify(params, {arrayFormat: 'brackets'})
  },
})
```

### Expected behavior

The parameters should be serialized, but it didn't

### Axios Version

v0.28.0

**Comment by AlexSanin** [comment]
any workaround for this?

**Comment by briwa** [comment]
I think we can close this since #6263 is merged and v0.28.1. is released. See: 3021e0ddbdeb5cd40e9e296851a8bd8ff45116d3
